kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkConfig.java

```diff
@@ -343,6 +367,34 @@
             TABLE_CREATE_DOC
         );
   }
+    /**
+     * Throw an exception if the passed-in properties do not constitute a valid sink.
+     * @param props sink configuration properties
+     */
+    public static void validate(Map<String, String> props) {
```

Comment:
In`BigQuerySinkConnector.start()` and `BigQuerySinkTask.start()`, we should validate configurations before using them, like
```
BigQuerySinkConfig.validate(properties)
config = new BigQuerySinkConfig(properties);
```

Reply:
cc @skyzyx

Reply:
Might consider putting this in a [validate](https://kafka.apache.org/25/javadoc/org/apache/kafka/connect/connector/Connector.html#validate-java.util.Map-) method in the connector. The framework calls that method when new configs are submitted for a connector via the REST API and, if any errors are reported by the connector, rejects the config before it gets applied.

Reply:
Sounds good.

Reply:
Oh, it looks like this is actually just copied from the framework: https://github.com/apache/kafka/blob/fcf45e1fac88238b1d3dbcfa1f324674939706f3/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SinkConnectorConfig.java#L82-L99

Anyone know why we're doing validation in this class at all for the `topics` and `topics.regex` properties? The framework should be doing all of that for us.

Reply:
Yes. It is from SinkConnectorConfig. My requirement was to bring topics.regex support for BQ connector. kafka connect already has framework for topics.regex and validation logic which is implemented in SinkConnectorConfig but BigQuerySinkConfig extends from AbstractConfig.

Reply:
Right, so because the framework already has support for validating and using these config properties, why bring them into the config here? They don't have to be defined here; the framework will still detect that this is a sink connector and ensure that the properties provided by the user for the connector config will contain either `topics` or `topics.regex`.

Also, have we considered how this will work when the connector is configured with a topics regex but has table creation disabled? Think we might have to remove the preemptive check for existing tables entirely, or move it into the task class which can then check dynamically based on the calculated table name for records it receives (could also override [SinkTask::open](https://kafka.apache.org/25/javadoc/org/apache/kafka/connect/sink/SinkTask.html#open-java.util.Collection-) but that breaks if an SMT adjusts the name of the topic for an incoming record after it's read from Kafka).

Reply:
Now I understand the pain. I just created a ticket to make `BigQuerySinkConfig` extend from `SinkConnectorConfig`, https://github.com/wepay/kafka-connect-bigquery/issues/266
I will merge this PR. And we will do clean up in a separate PR.

Reply:
Why extend from `SinkConnectorConfig`? All the logic from that class is handled automatically by the framework, there's no need to inherit from it.